src/renderer/routes/test_sequencer_panel/components/DesignBar.tsx

```diff
@@ -0,0 +1,266 @@
+                  setIsCreateProjectModalOpen(true);
+                }}
+              >
+                <Route size={16} className="mr-2 stroke-muted-foreground" />
+                New Sequence
+              </DropdownMenuItem>
+              <DropdownMenuItem
+                onClick={() => {
+                  importSequences();
+                }}
+              >
+                <Import size={16} className="mr-2 stroke-muted-foreground" />
+                Import Sequence
+              </DropdownMenuItem>
+              <DropdownMenuItem disabled={true}>
+                <LayoutGrid
+                  size={16}
+                  className="mr-2 stroke-muted-foreground"
+                />
+                Sequence Gallery
+              </DropdownMenuItem>
+            </DropdownMenuContent>
+          </DropdownMenu>
+        )}
+        {/* Comming soon
+        <Button
+          data-testid="add-text-button"
+          className="gap-2"
+          variant="ghost"
+          disabled={true}
+        >
+          <HardDriveDownload size={20} className="stroke-muted-foreground" />
+          Load Test Profile From Cloud
+        </Button>
+        */}
+
+        <div className="grow" />
+
+        <code className="inline-flex items-center justify-center p-3 text-sm text-muted-foreground">
+          {" "}
+          Test {summary.numberOfTestRun + "/" + summary.numberOfTest}
+        </code>
+        <code className="inline-flex items-center justify-center p-3 text-sm text-muted-foreground">
+          {" "}
+          Sequence{" "}
+          {summary.numberOfSequenceRun + "/" + summary.numberOfSequence}
+        </code>
+
+        <CyclePanel />
+
+        <div
+          data-cy="app-status"
+          id="app-status"
+          className="text-md m-1 inline-flex h-8 w-40 items-center justify-center rounded-md border font-semibold text-primary transition-colors"
+        >
+          {mapStatusToDisplay[summary.status]}
+        </div>
+      </div>
+      <div className="py-1" />
+    </div>
+  );
+}
+
+// Helper functions
+
+const getNumberOfTest = (data: TestSequenceElement[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.type === "test") count++;
+  });
+  return count;
+};
+
+const getNumberOfTestRun = (data: TestSequenceElement[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (
+      elem.type === "test" &&
+      elem.status != "pending" &&
+      elem.status != "running"
+    )
+      count++;
+  });
+  return count;
+};
+
+const getNumberOfSequence = (data: TestSequenceContainer[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.runable) count++;
+  });
+  return count;
+};
+
+const getNumberOfSequenceRun = (data: TestSequenceContainer[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.runable && elem.status != "pending" && elem.status != "running")
+      count++;
+  });
+  return count;
+};
```

Comment:
too much repeating logic
use this `countWhere` function instead and define all of these functions in terms of `countWhere` by just changing what you pass to predicate
```ts
export function countWhere<T>(arr: T[], predicate: (t: T) => boolean): number {
  return arr.reduce((acc, t) => (predicate(t) ? acc + 1 : acc), 0);
}
```